### ISSUE TITLE
`use_g_gnuc_flag_enum` is too eager

### ISSUE THREAD

**Issue by fmuellner** [bilelmoussaoui/gobject-lint#60]
It seems to classify any enum with at least one explicitly set value as flag type.

One case where this is an issue is in combination with the `property_enum_convention` rule, which suggests dropping `PROP_0` in favor of `PROP_FIRST_PROP = 1`.

**Comment by fmuellner** [comment]
Oh, it is actually *really* eager:

```c
typedef enum {
  LIST_FLAGS_NONE     = 0,
  LIST_FLAGS_USER     = 1 << 0,
  LIST_FLAGS_SYSTEM   = 1 << 1,
  LIST_FLAGS_ENABLED  = 1 << 2,
  LIST_FLAGS_DISABLED = 1 << 3,
  LIST_FLAGS_ACTIVE   = 1 << 4,
  LIST_FLAGS_INACTIVE = 1 << 5,
  LIST_FLAGS_NO_PREFS = 1 << 6,
  LIST_FLAGS_NO_UPDATES = 1 << 7,
} G_GNUC_FLAG_ENUM ListFilterFlags;
```

gives

```
  >typedef enum {
    LIST_FLAGS_NONE     = 0,
    LIST_FLAGS_USER     = 1 << 0,
    LIST_FLAGS_SYSTEM   = 1 << 1,
  warning: Enum 'G_GNUC_FLAG_ENUM' appears to be a flags enum but is missing G_GNUC_FLAG_ENUM attribute
  rule: use_g_gnuc_flag_enum
```

**Comment by bilelmoussaoui** [comment]
I will have a look next weekend

**Comment by A6GibKm** [comment]
The ast does not pick up any flags for enums at the moment, that is why the rule is broken atm. I tried both putting the flag before or after the enum name.

**Comment by bilelmoussaoui** [comment]
We actually do, see https://github.com/bilelmoussaoui/goblint/blob/main/gobject-ast%2Fsrc%2Fmodel%2Ftypes%2Fenum_info.rs#L12 but maybe we are doing something wrong

**Comment by A6GibKm** [comment]
What I meant is that that `Vec` turns out empty.

**Comment by fmuellner** [comment]
FWIW, the issue from the [earlier comment](https://github.com/bilelmoussaoui/goblint/issues/60#issuecomment-4320417666) where `G_GNUC_FLAG_ENUM` isn't detected, and `goblint --only use_g_gnuc_flag_enum --fix` just keeps adding additional annotations is still the case.

**Comment by bilelmoussaoui** [comment]
Should be better now, could you please re-try ?

**Comment by fmuellner** [comment]
No change I'm afraid.

FWIW, my reproducer is https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/4193 - running goblint in `subprojects/extensions-tool` should come back clean, but the enum-flag rule keeps failing (including if you run with `--fix`)

**Comment by bilelmoussaoui** [comment]
Just tested with that file, should be resolved in main.

**Comment by fmuellner** [comment]
I can confirm that 🎉

**Comment by bilelmoussaoui** [comment]
let us close this then
